src/coord/src/persistence.rs

```diff
@@ -0,0 +1,192 @@
+                    e
+                );
+                return None;
+            }
+        };
+        tokio::spawn(async move { compacter.run().await });
+        let wals = match WriteAheadLogs::new(config.wals_path.clone()) {
+            Ok(wals) => wals,
+            Err(e) => {
+                error!(
+                    "encountered error while trying to initialize write-ahead log: {}",
+                    e
+                );
+                return None;
+            }
+        };
+
+        Some(Self {
+            wals,
+            compacter_tx,
+            wals_path: config.wals_path.clone(),
+            traces_path: config.traces_path.clone(),
+            disabled: false,
+        })
+    }
+
+    pub fn resume(&mut self, id: GlobalId) -> Option<Vec<Message>> {
+        if self.disabled {
+            return None;
+        }
+
+        let persisted_trace = match Trace::resume(id, &self.traces_path, &self.wals_path) {
+            Ok(trace) => trace,
+            Err(e) => {
+                error!(
+                    "encountered error while trying to restart table {} {}",
+                    id, e
+                );
+                self.disabled = true;
+                return None;
+            }
+        };
+
+        let messages = match persisted_trace.read() {
+            Ok(messages) => Some(messages),
+            Err(e) => {
+                error!(
+                    "encountered error trying to read from persisted table {} {}",
+                    id, e
+                );
+                None
+            }
+        };
+
+        self.compacter_tx
+            .send(CompacterMessage::Resume(id, persisted_trace))
+            .unwrap();
+
+        messages
+    }
+
+    pub fn write_progress(&mut self, timestamp: Timestamp) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.write_progress(timestamp) {
+            error!(
+                "encountered error trying to write progress at ts {}: {}",
+                timestamp, e
+            );
+            self.disabled = true;
+        }
+    }
+
+    pub fn create(&mut self, id: GlobalId) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.create(id) {
+            error!(
+                "encountered error creating table for relation {}: {}",
+                id, e
+            );
+            self.disabled = true;
+            return;
+        }
+        if let Err(e) = self.compacter_tx.send(CompacterMessage::Add(id)) {
+            debug!("compacter dropped, disabling WAL: {}", e);
+            self.disabled = true;
+        }
+    }
+
+    pub fn destroy(&mut self, id: GlobalId) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.destroy(id) {
+            error!(
+                "encountered error destroying table for relation {}: {}",
```

Comment:
I am hitting this under the following scenario:
1. create a table and insert some stuff
2. restart
3. issue DROP TABLE

At that point, the following is reported:

```
2021-03-01T06:45:09.336719Z ERROR coord::persistence: encountered error destroying table for relation u1: asked to delete wal for relation: User(1) that doesn't exist    
```

After this, further tables created are not persisted at all to disk, no table_wals subdirectories are created for them.

Reply:
The intent of the design is to make it so that if either subsystem (the write ahead log or the compacted traces) encounters an error it completely shuts down and does nothing further (trying to preserve materialize not crashing at all costs in this stage)

do you think a second log line like "no further updates will be persisted to disk" or something like that would make the intent more obvious?

Reply:
Yes, a more ominous error message would be nice, but I think what needs to happen is that further INSERTS should report and error on this and all other tables in the database. Some database object is either fully persisted (a normal table) or not (a e.g. temporary table), stopping persisting stuff in the middle is bound to cause a lot of grief.

Reply:
I can't reproduce this one anymore either - I think the most recent commit fixed this one as well? let me know if it repros for you as well?

Reply:
Upgraded the error message to be more ominous.

>  I think what needs to happen is that further INSERTS should report and error on this and all other tables in the database. Some database object is either fully persisted (a normal table) or not (a e.g. temporary table), stopping persisting stuff in the middle is bound to cause a lot of grief.

I agree with this but unfortunately will need to tackle this in the next pr as I need to checkpoint state and shift to other work for a bit.